internal/vehicle/carwings.go

```diff
@@ -57,52 +77,180 @@ func (v *CarWings) initSession() error {
 	v.session = &carwings.Session{
 		Region: v.region,
 	}
-
+	v.refreshKey = ""
 	return v.session.Connect(v.user, v.password)
 }
 
-// chargeState implements the api.Vehicle interface
-func (v *CarWings) chargeState() (float64, error) {
+// carwingsVehicleStatus holds the relevant data extracted from carwings API
+// on vehicle status request
+type carwingsVehicleStatus struct {
+	VehicleStatus struct {
+		SOC            float64
+		Range          int64
+		ChargingStatus carwings.ChargingStatus
+		PlugStatus     carwings.PluginState
+		ClimateStatus  struct {
+			active      bool
+			targetTemp  float64
+			outsideTemp float64
+		}
+		LastRefresh time.Time
+	}
+}
+
+func (v *CarWings) status() (res carwingsVehicleStatus, err error) {
+	// follow up requested refresh
+	if v.refreshKey != "" {
+		return v.refreshResult()
```

Comment:
does refreshResult not need a session? it's only created later on?

Reply:
Refreshkey can only be not empty, if there was already an active session that returned the key => session init has already been done.